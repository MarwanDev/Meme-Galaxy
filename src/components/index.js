export { default as Navbar } from './Navbar/Navbar';
export { default as Header } from './Header/Header';
export { default as Categories } from './Categories/Categories';
export { default as Layout } from './Layout/Layout';
export { default as Category } from './Category/Category';
