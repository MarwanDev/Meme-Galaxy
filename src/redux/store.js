import { configureStore } from "@reduxjs/toolkit";
import memesReducer from "./memesSlice/memesSlice";

const store = configureStore({
  reducer: {
    meme: memesReducer,
  },
});

export default store;
