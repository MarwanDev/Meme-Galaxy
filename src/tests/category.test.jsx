import { Category } from "../components";
import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";
import store from "../redux/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

describe("Category", () => {
  it("renders each category correctly", () => {
    const category = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Category />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(category).toMatchSnapshot();
  });
  it("renders Category component correctly", () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Category />
        </Provider>
      </BrowserRouter>
    );
  });
});
