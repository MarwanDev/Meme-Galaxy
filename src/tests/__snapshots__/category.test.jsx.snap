// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Category > renders each category correctly 1`] = `
<div
  href=""
  id={null}
  onClick=""
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "textDecoration": "none",
    }
  }
>
  <a
    className=""
    href="/details"
    id={null}
    onClick={[Function]}
    style={
      {
        "fontSize": 60,
        "textDecoration": "none",
      }
    }
  />
  <h6
    style={
      {
        "color": "#fff",
        "textDecoration": "none",
      }
    }
  />
</div>
`;
